packages/controls/src/widget_dom_events.ts

```diff
@@ -0,0 +1,275 @@
+                    // Keep track of the listeners we are attaching so that we can
+                    // remove them if needed.
+                    this._cache_listeners(event, view, handler)
+                    break
+                default:
+                    console.log('Not familiar with that message source')
+                    break
+            }
+        }
+    }
+
+    attach_listeners() {
+        // console.log("Attaching listeners")
+        let current_source = this.get('source')
+        _.each(current_source.views, (view_promise) => {
+            Promise.resolve(view_promise).then((view) => {
+                this._add_listeners_to_view(view)
+            })
+        })
+    }
+
+    _add_key_listener(event_type, view) {
+        // Key listeners should:
+        //     + Only fire when the mouse is over the element.
+        //     + Not propagate up to the notebook (because imagine you
+        //       press 'x' on your widget and cut the cell...not what
+        //       you probably want the user to experience)
+        //
+        // The approach here is to add a key listener on mouseenter and remove
+        // it on mouseleave.
+        let key_handler = (event) => {
+            // console.log('Key presses FTW!', event)
+            if (this.get('ignore_modifier_key_events') &&
+               _.contains(this.get('_modifier_keys'), event.key)) {
+                // If the key event is supposed to be ignored, then skip it.
+                return
+            }
+            this._send_dom_event(event)
+            // Need this (and useCapture in the listener) to prevent the keypress
+            // from propagating to the notebook.
+            event.stopPropagation()
+        }
+        // Last argument useCapture needs to be true to prevent the event from
+        // passing through to the notebook; also need to stopPropagation in key_handler.
+        let capture_event = true
+        let enable_key_listen = () => {document.addEventListener(event_type, key_handler, capture_event)}
+        let disable_key_listen = () => {document.removeEventListener(event_type, key_handler, capture_event)}
+        view.el.addEventListener('mouseenter', enable_key_listen)
+        view.el.addEventListener('mouseleave', disable_key_listen)
+        this._cache_listeners('mouseenter', view, enable_key_listen)
+        this._cache_listeners('mouseleave', view, disable_key_listen)
+    }
+
+
+    _dom_click(generating_view, event) {
+        // Get coordinates relative to the
+        let relative_xy = _get_position(generating_view, event)
+        event['relativeX'] = relative_xy.x
+        event['relativeY'] = relative_xy.y
+        if ('_array_xy' in generating_view) {
+            console.log("Hey, nice Image!")
+            let array_coords = generating_view['_array_xy'](event)
+            event['arrayX'] = array_coords.x
+            event['arrayY'] = array_coords.y
+        }
```

Comment:
If we expand this from just "wheel" to include "mousedown", "mousemove" and "mouseup" it solves the problem of the first mouse button events getting processed by the browser.